aztec/src/main/kotlin/org/wordpress/aztec/watchers/FullWidthImageElementWatcher.kt

```diff
@@ -30,18 +32,19 @@ class FullWidthImageElementWatcher(val aztecText: AztecText) : TextWatcher {
     }
 
     private fun normalizeEditingAroundImageSpans(count: Int, start: Int) {
-        if (!aztecText.isTextChangedListenerDisabled()) {
+        if (aztecText.text.getSpans(0, 0, FullWidthImageProcessingMarker::class.java).isEmpty()) {
```

Comment:
> Can you please explain why it's better to also check for isTextChangedListenerDisabled()

Sure! So, the rationale behind the disabling/enabling the "TextChangeListener" is to limit the reactions to text changes. All text listeners should check for the flag and refrain from reacting to the text change, unless it's paramount to perform the reaction.

We trust the caller of the disabling to know exactly what they're doing by wanting to bypass the listeners. An example of bypassing the flag is the `EndOfBufferMarkerAdder` which makes sure that the end-of-buffer-marker presence is always consistent. But even that should be revisited some day.

In our case here, with FullWidthImageElementWatcher, its functionality doesn't seem critical enough to circumvent the flag so, it should respect the flag. Do we have evidence/cases where it should perform its function no matter what? Let's expand if we do have such cases in mind.

That said, the whole notion of the listeners disable flag is "dangerous" and we better be always vigilant when using it.